shared/output/cloud/compute_small/lib/google/cloud/compute/v1/addresses/rest/service_stub.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Auto-generated by gapic-generator-ruby. DO NOT EDIT!
+
+require "gapic/rest"
+require "google/cloud/compute/v1/compute_small_pb"
+
+module Google
+  module Cloud
+    module Compute
+      module V1
+        module Addresses
+          module Rest
+            ##
+            # REST service stub for the Addresses service.
+            #
+            class ServiceStub
```

Comment:
General question: What goes in the `ServiceStub` class vs the `Client` class? Why not generate all the code directly into `Client` and not have this file at all?

Reply:
This way clients for Rest and Grpc look symmetrical. 

On another level, 
`client` is an interface to the user, it deals with various input data,  and (for cloud) error shape (which is a way of saying 'various output data') and configuration
service_stub is an interface to the `gapic-common` rest layer, it deals with encoding and decoding data to/from json (and that data goes to the correct parts of http request, which is GRPC transcoding)
(then the client_stub deals with wrapping Faraday)

so it's cleaner this way, I think

Reply:
Symmetricality is good, but it's it's only really important in the interface. The `Service` and `Stub` classes on the gRPC side are generated by gRPC, and are "internal" for the purposes of GAPIC. They are not part of the actual client interface that a user should care about, and indeed we don't generate any documentation for them. Indeed, the only reason they exist at all is that gRPC is generating an extra interface layer that GAPIC has chosen to hide because we don't think it's good enough. We don't have that complexity on the REST side, and I don't see why we need to add it.

I could also see an argument for separating out `ServiceStub` if it were reasonable to expect its lower-level interface will be used by something other than the `Client` class. But I can't think of a use case at this point, and I'm inclined to argue YAGNI.

Reply:
I would argue that it's still a cleaner separation of concerns and keeps the client class from trying to do too many things.

Reply:
Also I disagree that the symmetry is important only for the interface. 
- it is for us when we try to reason about the code we generate as well. 
- Some (most?) end-users will end up reading the client class. It would be easier for them if they don't have to take in everything at once and if they have to read both clients, they would be able to go 'oh, this one is basically the same as the other one'.

Reply:
Let's discuss during the Ruby fixit time on Monday.

Reply:
refactored into grpc_transcoding helper class